clippy_lints/src/misc_early.rs

```diff
@@ -25,7 +25,7 @@ declare_clippy_lint! {
     /// let { a: _, b: ref b, c: _ } = ..
     /// ```
     pub UNNEEDED_FIELD_PATTERN,
-    style,
+    pedantic,
```

Comment:
I agree with @gnzlbg , that this should probably even be a `restriction` lint, so that you have to opt-in.

Reply:
I can make that change if you request it.

Pedantic is described as
```
`clippy::pedantic` (lints which are rather strict, off by default)
```

Restriction is described as
```
The [lint list](https://rust-lang.github.io/rust-clippy/master/index.html) also contains "restriction lints", which are
for things which are usually not considered "bad", but may be useful to turn on in specific cases. These should be used
very selectively, if at all.
```

I'll switch it to restriction which seems to match this better.

Reply:
> which are usually not considered "bad"

I think this perfectly matches this lint 👍